### DISCUSSION TITLE
class component가 필요한 경우는?

### DISCUSSION THREAD

**Discussion by JUDONGHYEOK** [woowacourse-study/2022-woowahan-bansanghwe#44] [React]
함수형 컴포넌트에는 존재하지 않는 생명주기를 사용할 때 class Component를 사용한다.

- getSnapshotBeforeUpdate : 가장 마지막으로 렌더링된 결과가 DOM 등에 반영되기 전에 호출
- getDerivedStateFromError: 하위의 자손 컴포넌트에서 오류가 발생했을 때 호출, 매개변수로 오류를 전달받고, 갱신된 state 값을 반드시 반환해야 함
- componentDidCatch: 자손 컴포넌트에서 오류가 발생했을 때에 호출, `componentDidCatch()`는 “커밋” 단계에서 호출되므로, 부수 효과를 발생시켜도 됨

**Comment by InKyoJeong** [comment]
오오 필요한 경우가 있군요 이름이 어려워요. 그래서아직 클래스 컴포넌트를 쓰는 라이브러리도 있나보네요. 그냥 오래돼서 쓰는것도 있겠지만,,
